Minor changes to improve user experience

Joyphish opened the pull request:
1. make shape fill color more transparent. The existing fill color is too opaque and it was difficult to see the underlying image.
2. updated zoom ratio for extra wide or extra high images. This will improve cursor's constant location relative to the image. Otherwise users see some pictures jumping around when it is zoomed to be bigger than view window.
3. draw label outside of bbox. When it is drawn inside of the box, the label blocks image view. When edit the bbox, users often need to see the image at the bbox corners. Drawing label outside of the bbox can help user seeing the image clearly.
4. change cross hair line style to differentiate editing and drawing mode. This will provide a visual cue for the mode. Otherwise it is difficult to tell the 2 modes apart, other than reading the text on top of the view.

Thanks for your contribution!  
Please confirm that you agree to the [Contributor License Agreement (CLA)](../CLA.md) by checking the box below:

- [x] I have read and agree to the CLA.

CVHub520 commented on anylabeling/views/labeling/widgets/canvas.py:
```diff
@@ -1472,10 +1476,15 @@ def paintEvent(self, event):  # noqa: C901
 
         # Draw mouse coordinates
         if self.cross_line_show:
```
```python
        if self.cross_line_show:
            pen = QtGui.QPen(
                QtGui.QColor(self.cross_line_color),
                max(1, int(round(self.cross_line_width / Shape.scale))),
                Qt.DotLine if self.editing() else Qt.SolidLine,
            )
```

Joyphish commented on anylabeling/resources/resources.py:
Sorry, I am new to this and did not know how to exclude it. I will do some research and avoid doing this in the future.

CVHub520 commented on anylabeling/views/labeling/widgets/canvas.py:
```diff
@@ -1422,15 +1422,19 @@ def paintEvent(self, event):  # noqa: C901
                         bbox = shape.bounding_rect()
                     except IndexError:
                         continue
+                    # draw the labe text outside of the shape
```
Conflict with sekf.text display

CVHub520 commented on anylabeling/views/labeling/label_widget.py:
please remove the commented-out code in this PR. We prefer to keep the source clean and avoid unnecessary comments.

CVHub520 commented:
Hi @Joyphish,

Thanks for the updates! I appreciate your effort in addressing the feedback. However, I noticed that some of the changes don’t fully align with the suggestions I previously shared. To keep the project consistent and maintainable, please take another look and make sure the revisions follow the earlier guidance.

If the feedback is not addressed in the next update, I may have to close this PR. You’re welcome to resubmit it later with the necessary adjustments.

Thanks for your understanding and cooperation!

Best regards,
X-AnyLabeling Maintainer

Joyphish commented:
> Hi @Joyphish,
> 
> Thanks for the updates! I appreciate your effort in addressing the feedback. However, I noticed that some of the changes don’t fully align with the suggestions I previously shared. To keep the project consistent and maintainable, please take another look and make sure the revisions follow the earlier guidance.
> 
> If the feedback is not addressed in the next update, I may have to close this PR. You’re welcome to resubmit it later with the necessary adjustments.
> 
> Thanks for your understanding and cooperation!
> 
> Best regards, X-AnyLabeling Maintainer

Sorry that I am new to Github and not quite familiar with the process. You can do whatever you need to do or revise your main source by yourself. Let me take some time to figure out the correct process. Thank you.
